app/src/androidTest/java/com/sdp/movemeet/view/map/MiniMapFragmentTest.java

```diff
@@ -39,50 +40,61 @@
 import static androidx.test.platform.app.InstrumentationRegistry.getInstrumentation;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertThat;
 
 @RunWith(AndroidJUnit4.class)
 public class MiniMapFragmentTest {
-    private UiDevice uiDevice;
     private FirebaseAuth fAuth;
-    private FirebaseUser user;
 
 
     @Rule
     public GrantPermissionRule mRuntimePermissionRule = GrantPermissionRule.grant(Manifest.permission.ACCESS_FINE_LOCATION);
 
     @Rule
-    public FragmentTestRule<?, MiniMapFragment> fragmentTestRule =
-            FragmentTestRule.create(MiniMapFragment.class);
+    public ActivityScenarioRule<UploadActivityActivity> testRule = new ActivityScenarioRule<>(UploadActivityActivity.class);
+
 
     @Before
     public void setUp() throws InterruptedException {
         fAuth = FirebaseAuth.getInstance();
-        fAuth.signInWithEmailAndPassword("test@test.com", "password").addOnCompleteListener(new OnCompleteListener<AuthResult>() {
-            @Override
-            public void onComplete(@NonNull Task<AuthResult> task) {
-                user = fAuth.getCurrentUser();
-                uiDevice = UiDevice.getInstance(getInstrumentation());
-            }
-        });
-        sleep(2000);
+        fAuth.signInWithEmailAndPassword("test@test.com", "password");
```

Comment:
Good point! We should probably always use the same "test account" for running our tests in the future. This one seems appropriate!

Reply:
@Antho1426 Ideally we would be mocking the Authentication backend service to not have to log in every time

Reply:
That would be even better you're right!

Reply:
yes indeed !